### PR TITLE
Rebase vfs code to use Moby definition of copy driver

### PR DESCRIPTION
**Pull request by rhatdan**
This new version of copy will attempt to use reflinks
if the underlying file system supports reflink, which
should optimize the vfs storage driver for rootless
podman and buildah.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@nalind @vrothberg @giuseppe @rhvgoyal PTAL

**Comment by giuseppe** [comment]
I've tested it with Podman and XFS with reflinks enabled and it works as expected

LGTM

**Comment by vrothberg** [comment]
The build errors look really odd:
```
[...]
vendor/github.com/ostreedev/ostree-go/pkg/otbuiltin/init.go:59: cannot use r (type C.gboolean) as type C.int in argument to isOk
[...]
```

**Comment by giuseppe** [comment]
there is PR to address it: https://github.com/containers/storage/pull/247

but needs another re-vendoring as it doesn't include the fix for go 1.8

**Comment by rhatdan** [comment]
@giuseppe The strange thing here is that these tests no longer exists in moby code.  So I am not sure what is correct or not here.

**Comment by nalind** [comment]
LGTM

File: drivers/vfs/copy_linux.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package vfs
+
+import "github.com/containers/storage/drivers/copy"
+
+func dirCopy(srcDir, dstDir string) error {
+	return copy.DirCopy(srcDir, dstDir, copy.Content, false)
```
Comment (nalind) [inline drivers/vfs/copy_linux.go]: I'm not 100% that this should be `false` rather than `true`, but it's not a blocker.

**Pull request merged**